File: javascript/resource.js
var resource = {};
// base model contains client only stuff
resource.baseModel = 
{
    getSprite: function ()
    {
        return this._sprite;
    },
    getType: function ()
    {
        return this._type;
    }
}

// create a resource type
resource.createModel = function (data)
{
    var model = $.extend({}, resource.baseModel, data);
    // allow it to be instancised
    resource[model._type] = function ()
    {
        return model;
    }
}

/*
resource.grass = function ()
{
    this._type = "grass";
    this._colour = "green";
    this._sprite = "grass";
    this._movementCost = 1;

    this.getSprite = resource.getSprite;
    this.getType = resource.getType;

    return this;
}

resource.swamp = function ()
{
    this._type = "swamp";
    this._colour = "green";
    this._sprite = "swamp";
    this._movementCost = 3;

    this.getSprite = resource.getSprite;
    this.getType = resource.getType;

    return this;
}

resource.water = function()
{
    this._type = "water";
    this._colour = "blue";
    this._sprite = "water";
    this._movementCost = 8;

    this.getSprite = resource.getSprite;
    this.getType = resource.getType;

    return this;
}

resource.forest = function ()
{
    this._type = "forest";
    this._colour = "green";
    this._sprite = "forest";
    this._movementCost = 2;

    this.getSprite = resource.getSprite;
    this.getType = resource.getType;

    return this;
}

resource.mountain = function ()
{
    this._type = "mountain";
    this._colour = "#817e6d";
    this._sprite = "mountain";
    this._movementCost = 5;

    this.getSprite = resource.getSprite;
    this.getType = resource.getType;

    return this;
}
*/

File: App_Code/Models/Resource.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

/// <summary>
/// Summary description for Resources
/// </summary>
public class Resource : BaseType
{
    public string _type;
    public string _colour;
    public string _sprite;
    public int _movementCost;
    public int _defenseModifier;
    public int _attackModifier;

    public Resource()
    {
        ModelType = "Resource";
    }
}

public class Grass : Resource
{
    public Grass()
    {
        _type = "grass";
        _colour = "green";
        _sprite = "grass";
        _movementCost = 1;
        _defenseModifier = 0;
        _attackModifier = 0;
    }
}

public class Swamp : Resource
{
    public Swamp()
    {
        _type = "swamp";
        _colour = "green";
        _sprite = "swamp";
        _movementCost = 3;
        _defenseModifier = 5;
        _attackModifier = -1;
    }
}

public class Water : Resource
{
    public Water()
    {
        _type = "water";
        _colour = "blue";
        _sprite = "water";
        _movementCost = 8;
        _defenseModifier = 0;
        _attackModifier = -5;
    }
}

public class Forest : Resource
{
    public Forest()
    {
        _type = "forest";
        _colour = "green";
        _sprite = "forest";
        _movementCost = 2;
        _defenseModifier = 3;
        _attackModifier = 3;
    }
}

public class Mountain : Resource
{
    public Mountain()
    {
        _type = "mountain";
        _colour = "#817e6d";
        _sprite = "mountain";
        _movementCost = 5;
        _defenseModifier = 5;
        _attackModifier = 2;
    }
}